WordPressKit/WordPressOrgXMLRPCApi.swift

```diff
@@ -59,18 +69,32 @@ open class WordPressOrgXMLRPCApi: NSObject {
         sessionConfiguration.httpAdditionalHeaders = additionalHeaders
         let sessionManager = Alamofire.SessionManager(configuration: sessionConfiguration)
 
-        let sessionDidReceiveChallengeWithCompletion: ((URLSession, URLAuthenticationChallenge, @escaping(URLSession.AuthChallengeDisposition, URLCredential?) -> Void) -> Void) = { [weak self] session, authenticationChallenge, completionHandler in
-            self?.urlSession(session, didReceive: authenticationChallenge, completionHandler: completionHandler)
+        let sessionDidReceiveChallengeWithCompletion: ((URLSession, URLAuthenticationChallenge, @escaping(URLSession.AuthChallengeDisposition, URLCredential?) -> Void) -> Void) = { [sessionDelegate] session, authenticationChallenge, completionHandler in
+            sessionDelegate.urlSession(session, didReceive: authenticationChallenge, completionHandler: completionHandler)
         }
         sessionManager.delegate.sessionDidReceiveChallengeWithCompletion = sessionDidReceiveChallengeWithCompletion
 
-        let  taskDidReceiveChallengeWithCompletion: ((URLSession, URLSessionTask, URLAuthenticationChallenge, @escaping(URLSession.AuthChallengeDisposition, URLCredential?) -> Void) -> Void) = { [weak self] session, task, authenticationChallenge, completionHandler in
-            self?.urlSession(session, task: task, didReceive: authenticationChallenge, completionHandler: completionHandler)
+        let  taskDidReceiveChallengeWithCompletion: ((URLSession, URLSessionTask, URLAuthenticationChallenge, @escaping(URLSession.AuthChallengeDisposition, URLCredential?) -> Void) -> Void) = { [sessionDelegate] session, task, authenticationChallenge, completionHandler in
+            sessionDelegate.urlSession(session, task: task, didReceive: authenticationChallenge, completionHandler: completionHandler)
         }
         sessionManager.delegate.taskDidReceiveChallengeWithCompletion = taskDidReceiveChallengeWithCompletion
         return sessionManager
     }
 
+    private func makeSession(configuration sessionConfiguration: URLSessionConfiguration) -> URLSession {
+        var additionalHeaders: [String: AnyObject] = ["Accept-Encoding": "gzip, deflate" as AnyObject]
+        if let userAgent = self.userAgent {
+            additionalHeaders["User-Agent"] = userAgent as AnyObject?
+        }
+        sessionConfiguration.httpAdditionalHeaders = additionalHeaders
+        return URLSession(configuration: sessionConfiguration, delegate: sessionDelegate, delegateQueue: nil)
+    }
+
+    // swiftlint:disable weak_delegate
+    /// `URLSessionDelegate` for the URLSession instances in this class.
+    private let sessionDelegate = SessionDelegate()
+    // swiftlint:enable weak_delegate
```

Comment:
swiftlint throws a warning/error on this simply because it sees a variable with a type that has "Delegate" in it is not declared as a weak reference. It doesn't really know how the delegate is supposed to be used, or if there is a real retain cycle. In this case here, swiftlint reports a false positive. So, it's okay to disable the rule on this particular variable.

`URLSessionDelegate` is a weird one: [`URLSession` holds a strong reference to its delegate](https://developer.apple.com/documentation/foundation/urlsession#2934758). So, we can't make `WordPressOrgXMLRPCApi`, which holds a strong reference to `URLSession` instances, implements `URLSessionDelegate`.

Technically, we can declare this variable as a weak reference, which will put the responsiblity of not getting it deallocated to the `URLSession` instances. I used a strong reference here, simply because I don't see any harm in it, which IMO seems more secure because we do need it during the whole lifetime of `WordPressOrgXMLRPCApi`.